ui/pages/onboarding-flow/create-password/create-password.test.js

```diff
@@ -0,0 +1,391 @@
+import React from 'react';
+import { fireEvent, waitFor } from '@testing-library/react';
+import configureMockStore from 'redux-mock-store';
+import { renderWithProvider } from '../../../../test/lib/render-helpers';
+import initializedMockState from '../../../../test/data/mock-send-state.json';
+import {
+  ONBOARDING_SECURE_YOUR_WALLET_ROUTE,
+  ONBOARDING_COMPLETION_ROUTE,
+} from '../../../helpers/constants/routes';
+import { FIRST_TIME_FLOW_TYPES } from '../../../helpers/constants/onboarding';
+import CreatePassword from './create-password';
+
+const mockHistoryPush = jest.fn();
+const mockHistoryReplace = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    push: mockHistoryPush,
+    replace: mockHistoryReplace,
+  }),
+}));
+
+describe('Onboarding Create Password', () => {
+  const mockState = {
+    metamask: {
+      identities: {},
+      selectedAddress: '',
+    },
+  };
+
+  const mockCreateNewAccount = jest.fn().mockResolvedValue();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Initialized State Conditionals with keyrings and firstTimeFlowType', () => {
+    it('should route to secure your wallet when keyring is present but not imported first time flow type', () => {
+      const mockStore = configureMockStore()(initializedMockState);
+
+      renderWithProvider(<CreatePassword />, mockStore);
+      expect(mockHistoryReplace).toHaveBeenCalledWith(
+        ONBOARDING_SECURE_YOUR_WALLET_ROUTE,
+      );
+    });
+
+    it('should route to completion when keyring is present and imported first time flow type', () => {
+      const importFirstTimeFlowState = {
+        ...initializedMockState,
+        metamask: {
+          ...initializedMockState.metamask,
+          firstTimeFlowType: 'import',
+        },
+      };
+      const mockStore = configureMockStore()(importFirstTimeFlowState);
+
+      renderWithProvider(<CreatePassword />, mockStore);
+      expect(mockHistoryReplace).toHaveBeenCalledWith(
+        ONBOARDING_COMPLETION_ROUTE,
+      );
+    });
+  });
+
+  describe('Render', () => {
+    it('should match snapshot', () => {
+      const mockStore = configureMockStore()(mockState);
+      const { container } = renderWithProvider(<CreatePassword />, mockStore);
+
+      expect(container).toMatchSnapshot();
```

Comment:
Can you validate the snapshot for button text for 
```
            {secretRecoveryPhrase &&
            firstTimeFlowType === FIRST_TIME_FLOW_TYPES.IMPORT
              ? t('importMyWallet')
              : t('createNewWallet')}
```

You can use matchInlineSnapshot :)

Reply:
Is this what you had in mind, [a91f538](https://github.com/MetaMask/metamask-extension/pull/18257/commits/a91f53876d59a84bca7f095013de721ed4972d78)?

Reply:
Yes, almost. You got the idea. 
it can be simplified as “button.textContent().matchInlineSnapshot()”